### PR TITLE
[core framework, experimental framework] Add `Account` module, support layered accounts, remove unneeded Move modules

### PR DESCRIPTION
**Pull request by tzakian**
This is a larger PR that adds the core account module and adds the ability to layer accounts on top of this core account. While large, this PR is smaller than you think, since a lot of the changes are in auto-generated files. 

To get a sense of the Move changes, I would suggest just looking at the updated [`core`](https://github.com/tzakian/libra/tree/layered_accounts/diem-move/diem-framework/core/sources) and [`experimental`](https://github.com/tzakian/libra/tree/layered_accounts/diem-move/diem-framework/experimental/sources) frameworks directly. After that you should be able to filter all files out except `.rs` when reviewing.

At a high level, this PR does the following:
* It introduces the `Account` module. This manages an accounts sequence number + auth key. At genesis a chain-specific type marker is registered. The prologue and epilogue can only be called by modules that can produce a witness of this type marker. The rules about who can access this type marker is encoded in the chain-specific code/framework.
* It creates a minimal core framework package. This does not yet contain configs, but Emma's work will be moving the core configs into this package.
* At genesis, the `ChainSpecificAccountInfo` resource is published under the `CoreResources` address. This provides information to the adapter about what functions to call for the prologue/epilogues for various transaction types. The values in this resource are chain-specific.
* This refactors the experimental framework to only rely on the core framework and removes all unneeded Move modules.
* The DPN framework only relies on the Move Stdlib now
* The Diem adapter has been updated to get the prologues/epilogues from the `ChainSpecificAccountInfo` resource if present. If this resource is not present it defaults to the DPN-specific prologues/epilogues.
* In Rust, we rename `AccountResource` to `DiemAccountResource`, I have moved as many places over to using the `AccountResource` as opposed to the `DiemAccountResource` wherever possible.
* Added some simple tests to make sure that the correct prologue is run based on the `ChainSpecificAccountInfo`, and that errors raised from the prologue are propagated correctly.
* Removed some tests in nft-forge since they were relying on DPN functionality (e.g., p2p payments and vasps)

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Changing JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Review by wrwg — APPROVED** [review]

**Review by emmazzz — APPROVED** [review]
Skimmed through the changes and overall it looks great! So happy to see all the file deletions 😄  Let's get this in and iterate.

File: diem-move/diem-framework/core/sources/Account.move [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    use Std::Signer;
+    use Std::Hash;
+    use Std::Vector;
+    use CoreFramework::ChainId;
+
+    /// Resource representing an account.
+    struct Account has key, store {
+        authentication_key: vector<u8>,
+        sequence_number: u64,
+        self_address: address,
```
[the remark this answers is not in the document]
Comment (tzakian) [reply diem-move/diem-framework/core/sources/Account.move]: Yes! There's this lovely function/comment in the account state:
```
// By design and do not remove
    pub fn get_account_address(&self) -> Result<Option<AccountAddress>>
```

So figured it was best not to mess with it....we can iterate on it and try to remove it later, but my thinking is it's best not to try and figure that out in this PR probably

**Pull request merged**